### ISSUE TITLE
feature request: Rust SDK

### ISSUE THREAD

**Issue by convivitybackendteam** [logto-io/logto#3767]
As part of our project development, we are in need of a reliable authentication (AUTH) solution. After conducting a thorough search, we came across Logto, and we were impressed with their approach to solving various authentication-related challenges.

However, we noticed that there is currently no software development kit (SDK) available for the Rust programming language. We believe that integrating Logto's AUTH solution with Rust will offer our team a streamlined and effective approach to securing our application.

We appreciate Logto's commitment to delivering clean and functional solutions and we would be thrilled to see an SDK made available for Rust in the near future. Thank you for considering our feature request.

**Comment by gao-sun** [comment]
thanks for your kind words and letting us know the request! we'll discuss about this. meanwhile, you are more than welcome if you are interested in contributing.

**Comment by AnthonyMichaelTDM** [comment]
I would also find this immensely useful, and would probably be willing/able to, at the very least start, the implementation. How would I get started?

**Comment by nbrugger-tgm** [comment]
Since logto behind the scenes is "just" a managed Oauth2 server is it possible to use logo from any language using a "generic" OAuth2/OCID library? Maybe it would be valuable to have a guide besides the custom SDKs (Vue, Java, PhP...) that is labeled as "Others" and shows how to use as a generic OAuth server. Then rust usage would be possible without a dedicated SDK right? Or are the SDKs more than just OAuth wrappers

**Comment by caioluis** [comment]
@gao-sun may I be assigned to this issue? I have already started my unofficial implementation, I'd love to contribute with other members as well.

**Comment by gao-sun** [comment]
> @gao-sun may I be assigned to this issue? I have already started my unofficial implementation, I'd love to contribute with other members as well.

Sure please go ahead. We can link to your repo once completed.

**Comment by gao-sun** [comment]
> Since logto behind the scenes is "just" a managed Oauth2 server is it possible to use logo from any language using a "generic" OAuth2/OCID library? Maybe it would be valuable to have a guide besides the custom SDKs (Vue, Java, PhP...) that is labeled as "Others" and shows how to use as a generic OAuth server. Then rust usage would be possible without a dedicated SDK right? Or are the SDKs more than just OAuth wrappers

Yes you are right. A guide for using existing OAuth or OIDC SDKs would be helpful indeed.

**Comment by gao-sun** [comment]
> I would also find this immensely useful, and would probably be willing/able to, at the very least start, the implementation. How would I get started?

You can get started with https://docs.logto.io/docs/references/sdk-convention/

**Comment by BjornTheProgrammer** [comment]
I have begun on a new crate that support axum, actix-web, and rocket by deafult feel free to check it out here [logto-sdk](https://github.com/BjornTheProgrammer/logto-sdk).
